File: app/src/main/java/com/example/android/test1/WeatherPresenterImpl.java
package com.example.android.test1;

import com.example.android.test1.POJO.WeatherDB;
import com.hannesdorfmann.mosby.mvp.MvpBasePresenter;

import java.util.List;

public class WeatherPresenterImpl extends MvpBasePresenter<IMainWeatherView> implements IWeatherPresenter, WeatherModel.GettingData {

    private final WeatherModel weatherModel;
    private boolean buttonPressed = false;

    public WeatherPresenterImpl() {
        weatherModel = new WeatherModel(this);
    }

    @Override
    public void buttonPressed() {
        buttonPressed = true;
        weatherModel.getAll();
    }

    @Override
    public void getAll(List<WeatherDB> weatherDBList) {
        if (buttonPressed) {
            if (isViewAttached()) {
                getView().showList(weatherDBList);
                buttonPressed = false;
            }
        }
    }
}

File: app/src/main/java/com/example/android/test1/MVP/WeatherPresenterImpl.kt
package com.example.android.test1.MVP

import com.hannesdorfmann.mosby.mvp.MvpBasePresenter
import com.example.android.test1.MVP.IMainWeatherView
import com.example.android.test1.MVP.IWeatherPresenter
import com.example.android.test1.MVP.WeatherModel
import com.example.android.test1.data.WeatherDB

class WeatherPresenterImpl : MvpBasePresenter<IMainWeatherView?>(), IWeatherPresenter {

    private val weatherModel: WeatherModel = WeatherModel()
    private var buttonPressed = false

    override fun buttonPressed() {
        buttonPressed = true
        Thread {
            val list = weatherModel.all
            view!!.showList(list)
        }.start()
    }
}